src/dotnet/commands/dotnet-build/Program.cs

```diff
@@ -93,10 +104,53 @@ public static int Run(string[] args)
 
                 msbuildArgs.AddRange(app.RemainingArguments);
 
-                return new MSBuildForwardingApp(msbuildArgs).Execute();
+                return 0;
             });
 
-            return app.Execute(args);
+            int exitCode = app.Execute(args);
+            if (msbuildArgs == null)
```

Comment:
@livarcocc I had that in my first initial version but then I thought that this will depend on internal behavior of command line parser - i.e. if parser decides to return 0 on `--help` then this will fail. I talked about that with @jonsequitur and we think this problem will completely disappear once we start refactoring logic around command line parser